django/test/client.py

```diff
@@ -458,11 +458,7 @@ def _encode_json(self, data, content_type):
         return json.dumps(data, cls=self.json_encoder) if should_encode else data
 
     def _get_path(self, parsed):
-        path = parsed.path
-        # If there are parameters, add them
-        if parsed.params:
-            path += ";" + parsed.params
-        path = unquote_to_bytes(path)
+        path = unquote_to_bytes(parsed.path)
```

Comment:
I'm a bit concerned about this change because the *original* logic is untested. In `main`, if you fully remove the two lines that re-adds params:

```python
        # If there are parameters, add them
        if parsed.params:
            path += ";" + parsed.params
```

No test fail. Would you be willing to have a separated, initial commit that adds a test for this logic?

Reply:
From my testing, I was able to make this code fail unit tests when it was configured incorrectly, although our cases might have been different. Given this is the test client, and thus quite important to be well-tested, I'll have a check through and see if there's some more opportunity to harden said tests.

Reply:
Thanks @RealOrangeOne! Given your comment I retried my test, using a fully clean `main` checkout and I think you are right. With this change:
```diff
diff --git a/django/test/client.py b/django/test/client.py
index aa42c1f60a..49d616b1d0 100644
--- a/django/test/client.py
+++ b/django/test/client.py
@@ -460,8 +460,8 @@ class RequestFactory:
     def _get_path(self, parsed):
         path = parsed.path
         # If there are parameters, add them
-        if parsed.params:
-            path += ";" + parsed.params
+        # if parsed.params:
+        #     path += ";" + parsed.params
         path = unquote_to_bytes(path)
         # Replace the behavior where non-ASCII values in the WSGI environ are
         # arbitrarily decoded with ISO-8859-1.
```
There is at least one unit test that fails:
```
======================================================================
FAIL: test_url_parameters (test_client.tests.ClientTest.test_url_parameters)
Make sure that URL ;-parameters are not stripped.
----------------------------------------------------------------------
Traceback (most recent call last):
...
AssertionError: '/unknown_view/' != '/unknown_view/;some-parameter'
- /unknown_view/
+ /unknown_view/;some-parameter
```
So I believe we are covered in this regard. Thanks for checking nevertheless!